sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -101,6 +101,8 @@ object ScalaReflection extends ScalaReflection {
       case t if t <:< definitions.ShortTpe => classOf[Array[Short]]
       case t if t <:< definitions.ByteTpe => classOf[Array[Byte]]
       case t if t <:< definitions.BooleanTpe => classOf[Array[Boolean]]
+      case t if t <:< localTypeOf[CalendarInterval] => classOf[Array[CalendarInterval]]
+      case t if t <:< localTypeOf[Decimal] => classOf[Array[Decimal]]
```

Comment:
both of them should be internal and won't be public to users, why do we change it?